### PR TITLE
Warn user if cargo ament build is not installed

### PR DESCRIPTION
**Pull request by luca-della-vedova**
This PR returns an error if `cargo-ament-build` was not found, including a message on how to install it, it's pretty straightforward but a few notes:

* I opted to recommend for using `cargo install cargo-ament-build` as noted in the [ros2-rust README](https://github.com/ros2-rust/ros2_rust), I thought about recommending pip but then it would cause a lot of trouble on 24.04 because virtual envs etc.
* I thought of falling back to a `colcon-cargo` package but I'm not 100% convinced it is a good idea to silently continue building with a build type different from what the users might expect, instead of loudly printing an error and failing.
* The failure method of this package identification is a bit odd and I'm not 100% sure it is the right thing to do, specifics below in the `Test it!` section.

### Test it!

* Copy the [sample package](https://github.com/colcon/colcon-ros-cargo/tree/main/test/rust-sample-package) into a colcon workspace
* Try a colcon build with `cargo ament-build` installed
* Remove `cargo-ament-build`
* Try again

You should see the (as I mentioned above slightly strange) failure:

```
lucadv@ros:~/colcon_ws$ colcon build
[0.241s] ERROR:colcon.colcon_ros_cargo.package_identification.ament_cargo:ament_cargo package found but cargo ament-build was not detected. Please install it by running: `cargo install cargo-ament-build`
[0.332s] WARNING:colcon.colcon_core.verb:No task extension to 'build' a 'ros.ament_cargo' package
```

Specifically the second warning is a bit odd but it will happen in all our error paths, so we should probably do something more than early returning on failure, but that is something out of scope for this specific PR

### CODE REVIEW

File: colcon_ros_cargo/package_identification/ament_cargo.py [hunk 1]
```diff
@@ -43,6 +45,14 @@ def identify(self, metadata):  # noqa: D102
                 'Got build type ament_cargo but could not find "Cargo.toml"')
             return
 
+        ament_build = 'cargo ament-build --help'.split()
+        if subprocess.run(ament_build, capture_output=True).returncode != 0:
+            logger.error(
+                'ament_cargo package found but cargo ament-build was not '
+                'detected. Please install it by running: '
+                '`cargo install cargo-ament-build`')
```
Comment (mxgrey) [inline colcon_ros_cargo/package_identification/ament_cargo.py]: This is a super minor nitpick but I wonder if we should tweak this line to
```
'\n $ cargo install cargo-ament-build'
```
Putting the solution on its own line and beginning it with a ` $ ` may help users notice the command and immediately clue them in that it's a recommendation to run a command.

I might also be a little inclined to `\n` at the start of `Please install it by running:` but that's less important.

**Review by mxgrey — APPROVED** [review]
Using the `--help` flag to test if the program exists is very clever.

I left a very minor recommendation for tweaking the way the information is presented, but I think the PR is good either way.

File: colcon_ros_cargo/package_identification/ament_cargo.py (hunk 1 above)
Comment (luca-della-vedova) [reply colcon_ros_cargo/package_identification/ament_cargo.py]: Tried that out, before (without the newline):

image

After (with the newline):

image

I have to say I don't actually like the newline approach, colcon logging adds a header to the line specifyin where / when the error originated, the moment we add a newline that header is lost and the output might be confused with the rest of the command's output.
A probably better option is to just add a new `logger` call so the header is included, but this also doesn't look terribly clean:

image

What do you think?

Comment (mxgrey) [reply colcon_ros_cargo/package_identification/ament_cargo.py]: The header really does add a lot of noise and makes the warning easy to overlook.

Here's a PR based on my own experimental iteration on how to display the warning: https://github.com/luca-della-vedova/colcon-ros-cargo/pull/4

**Review by mxgrey — APPROVED** [review]
Looks good, I think this will really help users to navigate the unusual Rust dependency.

**Review by esteve — APPROVED** [review]
@luca-della-vedova thanks!

**Pull request merged**